src/v/redpanda/admin_server.cc

```diff
@@ -1343,12 +1379,106 @@ void admin_server::register_features_routes() {
           auto version = ft.get_active_version();
 
           res.cluster_version = version;
-          for (const auto& f : ft.get_active_features()) {
-              res.features.push(ss::sstring(cluster::to_string_view(f)));
+          for (const auto& fs : ft.get_feature_state()) {
+              ss::httpd::features_json::feature_state item;
+              vlog(
+                logger.trace,
+                "feature_state: {} {}",
+                fs.spec.name,
+                fs.get_state());
+              item.name = ss::sstring(fs.spec.name);
+
+              switch (fs.get_state()) {
+              case cluster::feature_state::state::active:
+                  item.state = ss::httpd::features_json::feature_state::
+                    feature_state_state::active;
+                  break;
+              case cluster::feature_state::state::unavailable:
+                  item.state = ss::httpd::features_json::feature_state::
+                    feature_state_state::unavailable;
+                  break;
+              case cluster::feature_state::state::available:
+                  item.state = ss::httpd::features_json::feature_state::
+                    feature_state_state::available;
+                  break;
```

Comment:
it looks like `cluster::feature_state::state::available` aren't serialized anywhere, and the json endpoint spec defines the values for rest interface, so I'm wondering if the explicit values for `cluster::feature_state::state::available` (and others) are needed?

Reply:
yeah, the explicit values were redundant once I made the decision not to use those enums in serialized log messages -- removed.